backend-api/src/functions/asyncCredential/sessionService/sessionService.ts

```diff
@@ -26,26 +26,28 @@ export class SessionService implements IGetSessionBySub, ICreateSession {
 
   async getAuthSessionBySub(
     sub: string,
-    state: string,
     sessionRecoveryTimeout: number,
   ): Promise<ErrorOrSuccess<string | null>> {
     const queryCommandInput: QueryCommandInput = {
       TableName: this.tableName,
       IndexName: this.indexName,
       KeyConditionExpression:
-        "subjectIdentifier = :subjectIdentifier and issuedOn > :issuedOn",
-      FilterExpression:
-        "authSessionState = :authSessionState and #state <> :state",
-      ExpressionAttributeValues: {
-        ":subjectIdentifier": { S: sub },
-        ":authSessionState": { S: "ASYNC_AUTH_SESSION_CREATED" },
-        ":issuedOn": { S: (Date.now() - sessionRecoveryTimeout).toString() },
-        ":state": { S: state },
-      },
+        "#sub = :subVal and #sessionState = :sessionStateVal",
+      FilterExpression: "#issuedOn > :sessionExpirationThresholdMillis",
```

Comment:
This reads a bit strangely to me. 

It would be really helpful for this to say something like:

currentTime < sessionExpiry

Reply:
Updated